### PR TITLE
[util] change require fast-deep-equal to import react-fast-compare

### PR DESCRIPTION
**Pull request by bugzpodder**
# Description

I was having trouble consuming the esm build because of the require call.
I switched the packages because I ran into some ts definition errors when using fast-deep-equal.  
react-fast-compare is a fork of fast-deep-equal/react that handles edge cases a bit better.

## Type of change

- [x] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] **Breaking change** _(fix or feature that would cause existing functionality to not work as before)_

# Checklist
- [x] Read the [Contributing Guide](https://github.com/streamich/react-use/blob/master/CONTRIBUTING.md)
- [x] Perform a code self-review
- [ ] Comment the code, particularly in hard-to-understand areas
- [ ] Add documentation
- [ ] Add hook's story at Storybook
- [ ] Cover changes with tests
- [x] Ensure the test suite passes (`yarn test`)
- [x] Provide 100% tests coverage
- [x] Make sure code lints (`yarn lint`). Fix it with `yarn lint:fix` in case of failure.
- [x] Make sure types are fine (`yarn lint:types`).

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import reactFastCompare from 'react-fast-compare';
```
Comment (streamich) [inline src/util.ts:1]: Is it possible to do here?

```js
import isDeepEqual from 'fast-deep-equal/react`;
```

Comment (bugzpodder) [reply src/util.ts:1]: @streamich no, i got a bunch of typescript errors, let me dig it up

Comment (bugzpodder) [reply src/util.ts:1]:
```
(virtualenv3) jack-zhao@Colors-MBP react-use % yarn build:cjs
yarn run v1.22.4
$ tsc
node_modules/fast-deep-equal/react.d.ts:1:1 - error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.

1 const equal: (a: any, b: any) => boolean;
  ~~~~~
import reactFastCompare from 'react-fast-compare';

src/util.ts:1:8 - error TS2395: Individual declarations in merged declaration 'isDeepEqual' must be all exported or all local.

1 import isDeepEqual from 'fast-deep-equal/react';
         ~~~~~~~~~~~

src/util.ts:1:8 - error TS2440: Import declaration conflicts with local declaration of 'isDeepEqual'.

1 import isDeepEqual from 'fast-deep-equal/react';
         ~~~~~~~~~~~

src/util.ts:13:14 - error TS1155: 'const' declarations must be initialized.

13 export const isDeepEqual;
                ~~~~~~~~~~~

src/util.ts:13:14 - error TS2395: Individual declarations in merged declaration 'isDeepEqual' must be all exported or all local.

13 export const isDeepEqual;
                ~~~~~~~~~~~


Found 5 errors.
```

Comment (bugzpodder) [reply src/util.ts:1]: from https://github.com/FormidableLabs/react-fast-compare
`This version of react-fast-compare tracks fast-deep-equal@3.1.1.`

Comment (streamich) [reply src/util.ts:1]: [This should work](https://github.com/streamich/react-use/pull/1282/files#diff-8cd4968b81985f0efc2053eabc37db6dR1-R13) 🤞  once the latest changes of `fast-deep-equal` are published https://github.com/epoberezkin/fast-deep-equal/issues/69

**Comment by streamich** [comment]
This should hopefully fix it: https://github.com/streamich/react-use/pull/1281

**Pull request closed without merging**